### PR TITLE
Adds ability to reset field angle

### PR DESCRIPTION
**Pull request by penguin212**
The resetFieldAngle function resets the forward position for the joystick when it is called (not assigned to a button yet).

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/swerve/SwerveModule2020.java [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.kinematics.SwerveModuleState;
+
+import frc.robot.motorcontrol.MotorUtil;
+
+/**
+ * A swerve module with a NEO drive motor and a BAG steer motor, for running swerve modules on 
+ * the 2020 robot.
+ */
+public class SwerveModule2020 {
```
Comment (e3l) [inline src/main/java/frc/robot/subsystems/swerve/SwerveModule2020.java:24]: wait is this class duplicated

File: src/main/java/frc/robot/RobotContainer.java [hunk 2]
```diff
@@ -26,8 +28,8 @@
  */
 public class RobotContainer {
     // Subsystems
-    //private final SwerveSubsystem swerveSubsystem;
-    private final MissileShellSwerveSubsystem swerveSubsystem;
+    private final ISwerveSubsystem swerveSubsystem;
```
Comment (e3l) [inline src/main/java/frc/robot/RobotContainer.java:31]: small detail... i don't really like the "I" naming lol, esp cuz it's not an interface... what about AbstractSwerveSubsystem, perhaps

File: src/main/java/frc/robot/subsystems/swerve/SwerveSubsystem.java [hunk 3]
```diff
@@ -211,4 +214,13 @@ public void resetPosition() {
     private Rotation2d getGyroHeading() {
         return Rotation2d.fromDegrees(-ahrs.getAngle());
     }
+
+    private Rotation2d getGyroForField() {
+        return Rotation2d.fromDegrees(-(ahrs.getAngle() - angleoffset));
+    }
+
+    public void resetFieldAngle() {
```
Comment (e3l) [inline src/main/java/frc/robot/subsystems/swerve/SwerveSubsystem.java:222]: the final challenge is abstracting this code (that's common between the two years) too :))))

File: src/main/java/frc/robot/RobotContainer.java (hunk 2 above)
Comment (ky28059) [reply src/main/java/frc/robot/RobotContainer.java:31]: Yeah, as an abstract class `BaseSwerveSubsystem` makes more sense.

File: src/main/java/frc/robot/subsystems/swerve/SwerveSubsystem.java (hunk 3 above)
Comment (ky28059) [reply src/main/java/frc/robot/subsystems/swerve/SwerveSubsystem.java:222]: The states, gyro, and `setSwerveDrivePowers` can be abstracted too! Maybe even periodic? The hardest part would be abstracting the modules and adding odometry to 2020.

File: src/main/java/frc/robot/subsystems/swerve/SwerveModule2020.java (hunk 1 above)
Comment (ky28059) [reply src/main/java/frc/robot/subsystems/swerve/SwerveModule2020.java:24]: The difference between the 2020 swerve module and 2023 is that 2020 uses different motors and needs different PID constants to run. Also 2020 doesn't have the subclass setup in place yet, which we could consider adding.

File: src/main/java/frc/robot/subsystems/swerve/SwerveModule.java [hunk 4]
```diff
@@ -115,29 +118,32 @@ public SwerveModule(int drivePort, int steerPort) {
...
      */
     public SwerveModulePosition getState() {
         return new SwerveModulePosition(
-            // driveMotor.getSelectedSensorPosition() * DRIVE_TICKS_TO_METERS,
-            driveEncoder.getPosition(),
-            getAngle()
-        );
+                // driveMotor.getSelectedSensorPosition() * DRIVE_TICKS_TO_METERS,
+                driveEncoder.getPosition(),
+                getAngle());
```
Comment (ky28059) [inline src/main/java/frc/robot/subsystems/swerve/SwerveModule.java:128]: Oh you used format file didn't you :( now it'll make multi-line constructors and method invocations look ugly like this and break lines suboptimally

**Review by e3l — APPROVED** [review]
would recommend renaming `ISwerveSubsystem` to `AbstractSwerveSubsystem` or something similar.

Otherwise lgtm, can merge if you don't plan on more aggressively abstracting

**Review by ky28059 — APPROVED** [review]
👍

**Review by e3l — APPROVED** [review]
lgtm

**Pull request merged**